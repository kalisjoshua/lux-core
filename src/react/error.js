/**
 * @module Error
 * @memberof luxReact
 */

import React from 'react'; // `React` must be in scope when using JSX

// import Link from './link';
import registry from './componentRegistry';
import responseModelShape from './responseModel.propType';

function errorComponent(props) {

  return (
    <section>
      <h1>Error</h1>
      <p>Luxui encountered an error.</p>
      <p>{`${props.error}`}</p>
      <pre><code>{JSON.stringify(props.error, null, ' ')}</code></pre>
    </section>
  );
}
errorComponent.propTypes = responseModelShape;

registry('Error', errorComponent);
