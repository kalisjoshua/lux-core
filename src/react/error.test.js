import React from 'react';
import renderer from 'react-test-renderer';

import registry from './componentRegistry';

import './error';
const Component = registry('Error');

describe('Error - main - Component', function () {
  it('should exist; and should be a function', function () {
    expect(typeof Component).toBe('function');
  });

  it('should match the snapshot', function () {
    const response = {
      data: {},
      error: new Error('Unforseen error.'),
      path: '/',
      status: 400,
    };

    const component = renderer.create(
      <Component {...response} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
