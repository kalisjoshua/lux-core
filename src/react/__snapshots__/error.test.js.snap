exports[`Error - main - Component should match the snapshot 1`] = `
<section>
  <h1>
    Error
  </h1>
  <p>
    Luxui encountered an error.
  </p>
  <p>
    Error: Unforseen error.
  </p>
  <pre>
    <code>
      {}
    </code>
  </pre>
</section>
`;
